llvm/lib/CodeGen/RegisterClassInfo.cpp

```diff
@@ -142,7 +144,12 @@ void RegisterClassInfo::compute(const TargetRegisterClass *RC) const {
 
   // FIXME: Once targets reserve registers instead of removing them from the
   // allocation order, we can simply use begin/end here.
-  ArrayRef<MCPhysReg> RawOrder = RC->getRawAllocationOrder(*MF);
+  ArrayRef<MCPhysReg> RawOrder = RC->getRawAllocationOrder(*MF, Reverse);
+  std::vector<MCPhysReg> ReverseOrder;
+  if (Reverse) {
+    llvm::append_range(ReverseOrder, reverse(RawOrder));
```

Comment:
I probably miss something but why can't we define the correct reverse order in `AltOrders` so we don't need to reverse a list once again. We've already passed `Reverse=true` to `getRawAllocationOrder` and after this need to reverse once more.